server.go

```diff
@@ -1675,6 +1858,46 @@ func (w *responseWriter) Flush() {
 	}
 }
 
+type pushPromise struct {
+	reqpm requestParam
+	done  chan error
+}
+
+// Pusher is an interface that http.ResponseWriters implement if they support server push
+// When Push is called, it will send a push promise to the client, using the method, path and
+// headers. It will then initiate the push by calling the server-level http.Handler
+// for the path.
+type Pusher interface {
+	Push(method string, path string, h http.Header) error
+}
+
+func (w *responseWriter) Push(method string, path string, h http.Header) error {
+	rws := w.rws
+	if rws == nil {
+		panic("Push called after Handler finished")
+	}
+	if h == nil {
+		h = rws.req.Header
```

Comment:
I hope I understand this correctly... If the response handler that handles the initial request that comes from the client does a "Set-Cookie", the follow-up (pushed) request won't see the cookie, because the headers are simply copied from the initial request that comes from the client.

Should the cloned header include the cookie that was just set?